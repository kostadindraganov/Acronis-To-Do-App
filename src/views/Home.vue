<template>
  <el-row type="flex" justify="center" align="middle">
    <el-col :xs="24" :sm="22" :md="20" :lg="20" :xl="22">
      <el-card class="box-card" body-style="list-wrapper">
        <el-row :gutter="30" type="flex" justify="space-around" align="middle">
          <el-col :span="22">
            <el-input
              @keyup.enter.native="addNewItem"
              placeholder="Please input the activity"
              v-model="title"
              clearable
              type="text"
            ></el-input>
          </el-col>
          <el-col :span="2">
            <el-button
              type="primary"
              class="add-item-btn hidden-sm-only"
              :disabled="!title.length > 0"
              @click="addNewItem"
            >Add</el-button>
          </el-col>
        </el-row>
      </el-card>
      <transition name="el-zoom-in-top" v-if="currentList.length > 0">
        <div>
          <ListItem
            v-for="item in currentList"
            :key="item.id"
            :itemId="item.id"
            :itemTitle="item.title"
            :isDone="item.state"
          />
        </div>
      </transition>
      <transition name="el-zoom-in-bottom" v-else>
        <el-card v-loading="loading" shadow="hover" class="card-item-wrapper">
          <strong>No items</strong>
        </el-card>
      </transition>
    </el-col>
  </el-row>
</template>

<script>
  import { mapState } from "vuex";
  import { GET_CURRENT_LIST, GET_ADD_ITEM } from "@/store/mutation-types";
  import ListItem from "@/components/ListItem";

  export default {
    name: "home",
    data: () => ({
      title: "",
      loading: true
    }),
    components: {
      ListItem
    },
    computed: {
      ...mapState({
        currentList: state => state.currentList
      })
    },
    created() {
      this.$store.dispatch(GET_CURRENT_LIST).then(res => {
        this.loading = false;
      });
    },
    methods: {
      addNewItem() {
        if (this.title.length > 0) {
          this.loading = true;
          const newItem = { title: this.title, state: false };
          this.$store.dispatch(GET_ADD_ITEM, newItem).then(res => {
            this.loading = false;
            this.$store.dispatch(GET_CURRENT_LIST);
            this.title = "";
          });
        }
      }
    }
  };
</script>


