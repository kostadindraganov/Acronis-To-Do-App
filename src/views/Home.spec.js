import { config } from "@vue/test-utils";
import { createLocalVue, shallowMount, mount } from "@vue/test-utils";
import Vuex from "vuex";
import ElementUI from "element-ui";
import Home from "@/views/Home";

config.stubs.transition = false;
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(ElementUI);

describe("Home.vue", () => {
	let actions;
	let store;
	beforeEach(() => {
		actions = {
			getCurrentList: jest.fn(),
			getAddItem: jest.fn()
		};

		store = new Vuex.Store({
			state: {
				currentList: []
			},
			actions
		});
	});

	it("should dispatch action when created", () => {
		shallowMount(Home, {
			localVue,
			store
		});
		expect(actions.getCurrentList).toHaveBeenCalled();
	});

	it("dispatches an action new item is created", () => {
		const wrapper = mount(Home, {
			store,
			localVue
		});
		const textInput = wrapper.find('input[type="text"]');
		textInput.setValue("test");
		wrapper.find(".el-button").trigger("click");
		expect(wrapper.vm.title).toBe("test");

		expect(actions.getAddItem).toHaveBeenCalled();
	});

	test("renders correctly", () => {
		const wrapper = mount(Home, {
			localVue,
			store
		});
		expect(wrapper.element).toMatchSnapshot();
	});
});
