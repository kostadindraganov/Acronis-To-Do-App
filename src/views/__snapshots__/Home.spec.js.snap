// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders correctly 1`] = `
<div
  class="el-row is-justify-center is-align-middle el-row--flex"
>
  <div
    class="el-col el-col-24 el-col-xs-24 el-col-sm-22 el-col-md-20 el-col-lg-20 el-col-xl-22"
  >
    <div
      class="el-card box-card is-always-shadow"
    >
      <!---->
      <div
        class="el-card__body"
      >
        <div
          class="el-row is-justify-space-around is-align-middle el-row--flex"
          style="margin-left: -15px; margin-right: -15px;"
        >
          <div
            class="el-col el-col-22"
            style="padding-left: 15px; padding-right: 15px;"
          >
            <div
              class="el-input el-input--suffix"
            >
              <!---->
              <input
                autocomplete="off"
                class="el-input__inner"
                placeholder="Please input the activity"
                type="text"
              />
              <!---->
              <!---->
              <!---->
            </div>
          </div>
           
          <div
            class="el-col el-col-2"
            style="padding-left: 15px; padding-right: 15px;"
          >
            <button
              class="el-button add-item-btn hidden-sm-only el-button--primary is-disabled"
              disabled="disabled"
              type="button"
            >
              <!---->
              <!---->
              <span>
                Add
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="el-card card-item-wrapper is-hover-shadow"
    >
      <!---->
      <div
        class="el-card__body"
      >
        <strong>
          No items
        </strong>
      </div>
    </div>
  </div>
</div>
`;
